Src/WindowsServer/WindowsServer.Shared/Implementation/DataContracts/AzureInstanceComputeMetadata.cs

```diff
@@ -0,0 +1,177 @@
+﻿namespace Microsoft.ApplicationInsights.WindowsServer.Implementation.DataContracts
+{
+    using System;
+    using System.Globalization;
+    using System.Linq;
+    using System.Runtime.Serialization;
+    using System.Text.RegularExpressions;
+
+    /// <summary>
+    /// Class representing the returned structure from an Azure Instance Metadata request
+    /// for Compute information.
+    /// </summary>
+    [DataContract]
+    internal class AzureInstanceComputeMetadata
+    {
+        private static int resourceGroupNameLengthMax = 90;
+        private static int resourceGroupNameLengthMin = 1;
+        private static string resourceGroupNameValidChars = @"^[a-zA-Z0-9\.\-_]+$";
+        private static int nameLenghtMax = 64; // 15 for windows, go with Linux for MAX
+        private static int nameLengthMin = 1;
+        private static string nameValidChars = @"^[a-zA-Z0-9()_\-]+$";
+        private static TimeSpan regexTimeout = TimeSpan.FromMilliseconds(1000);
+
+        [DataMember(Name = "osType", IsRequired = true)]
+        internal string OsType { get; set; }
+
+        [DataMember(Name = "location", IsRequired = true)]
+        internal string Location { get; set; }
+
+        [DataMember(Name = "name", IsRequired = true)]
+        internal string Name { get; set; }
+
+        [DataMember(Name = "offer", IsRequired = true)]
+        internal string Offer { get; set; }
+
+        [DataMember(Name = "platformFaultDomain", IsRequired = true)]
+        internal string PlatformFaultDomain { get; set; }
+
+        [DataMember(Name = "platformUpdateDomain", IsRequired = true)]
+        internal string PlatformUpdateDomain { get; set; }
+
+        [DataMember(Name = "publisher", IsRequired = true)]
+        internal string Publisher { get; set; }
+
+        [DataMember(Name = "sku", IsRequired = true)]
+        internal string Sku { get; set; }
+
+        [DataMember(Name = "version", IsRequired = true)]
+        internal string Version { get; set; }
+
+        [DataMember(Name = "vmId", IsRequired = true)]
+        internal string VmId { get; set; }
+
+        [DataMember(Name = "vmSize", IsRequired = true)]
+        internal string VmSize { get; set; }
+
+        [DataMember(Name = "subscriptionId", IsRequired = true)]
+        internal string SubscriptionId { get; set; }
+
+        [DataMember(Name = "resourceGroupName", IsRequired = true)]
+        internal string ResourceGroupName { get; set; }
+
+        internal string GetValueForField(string fieldName)
```

Comment:
why 2 separate methods? can we just combine the fetching logic and sanitize logic into one?

Reply:
I would argue that couples the two different concerns a bit too much.